main/solution/backend/config/settings/.defaults.yml

```diff
@@ -61,6 +61,19 @@ apiHandlerRoleArn: 'arn:aws:iam::${self:custom.settings.awsAccountInfo.awsAccoun
 # env launch options)
 envTypeConfigsBucketName: ${self:custom.settings.globalNamespace}-env-type-configs
 
+# The S3 bucket name to be used to store egress store data
+egressStoreBucketName: ${self:custom.settings.globalNamespace}-egress-store
+
+# The alias used for the KMS key created to encrypt/decrypt egress store
+egressStoreKmsKeyAlias: ${self:custom.settings.globalNamespace}/s3/egressstore
+
+# The arn of the egress store KMS Key alias
+egressStoreKmsKeyAliasArn: arn:aws:kms:${self:custom.settings.awsRegion}:${self:custom.settings.awsAccountInfo.awsAccountId}:alias/${self:custom.settings.egressStoreKmsKeyAlias}
+
+# The statement ID in the generated KMS key's key policy that controls which workspaces
+#   can decrypt data in the egress store data bucket
+egressStoreKmsPolicyWorkspaceSid: Allow workspace environments to CRUD egress data
```

Comment:
For some of these properties, isn't it best to manage and name them internally. Kind of think that this would give admins a chance to change kms key when it isn't really straightforward to change it if there are active workspaces in flight. Managing these resources internally without admins controlling the name would help us in avoiding such mistakes. The only flag that can be exposed is egress store being enabled or disabled, and we might want to make it irreversible to begin with.

Reply:
Totally agree with you that we should ONLY expose the flag regarding to enable or disable egress store. Those properties are exposed in backend/config/settings instead of main/config/settings, I am putting them there to offer more flexibility for customers during Path to Production. I think we might need separate discussion about how to decide if we need the property exposed.

Reply:
By offering these properties, it seems like we are also giving a false assumption that these values can be changed and changes would be backwards compatible which I don't think is the case. Since changing these properties will have weird side effects, I suggest that we don't really offer them as a configuration unless there is a hard requirement. If there is a hard requirement then we need to either make these properties immutable or we need to support the fact that they can change. We can have a separate discussion with some more folks tomorrow to settle this.

Reply:
We spoke offline. As you pointed out these settings are not part of the main config so keeping it here makes sense. We should just document that changing these settings can have undesirable consequences on egress feature so it isn't recommended that we change them. Looks good otherwise. Sorry about the unnecessary churn.

Reply:
Resolved in commit: 37134d5ad08fc2ca00375c88d859041d7fe4ee0d